libs/utils/android/workloads/camerapreview.py

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+
+from subprocess import Popen, PIPE
+from android import Screen, Workload
+from time import sleep
+from devlib.utils.android import adb_command
+
+import logging
+
+GET_FRAME_STATS_CMD = 'shell dumpsys gfxinfo {} > {}'
+
+class CameraPreview(Workload):
+    """
+    Android CameraPreview workload
+    """
+
+    # Package required by this workload
+    package = 'com.google.android.GoogleCamera'
+    activity = 'com.android.camera.CameraActivity'
+
+    # Setup logger
+    logger = logging.getLogger('CameraPreview')
+    logger.setLevel(logging.INFO)
+
+
+    def __init__(self, test_env):
+        super(CameraPreview, self).__init__(test_env)
+        logging.debug('%14s - Workload created', 'CameraPreview')
+
+    def run(self, exp_dir, preview_duration_s, collect=''):
+
+        # Initialize energy meter results
+        nrg_data, nrg_file = None, None
+
+        # Unlock device screen (assume no password required)
+        self.target.execute('input keyevent 82')
+        # Press Back button to be sure we run the video from the start
+        self.target.execute('input keyevent KEYCODE_BACK')
```

Comment:
There is a new Android::System module available... it would be great if we can move these previous two commands to be a standard API of that module. Something like:
```
android.system.unlock_screen()
android.system.press_back_button()
```